prometheus/vnext.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+type v2 struct{}
+
+var V2 = v2{}
```

Comment:
Nice. Let's add some commentary on this exported variable. For example, the fact we are using this as a potential early experimentation of potential v2 API? Should it have an experimental mark on, so we can change it later or never in v1? 

I think I like the alias *I think* it makes it impossible for callers to change this variable globally which is great. I wonder why this and not `func V2() v2`?

Reply:
Good question... I guess mainly because I prefer the way `V2.bla()` looks compared to `V2().bla()`, but I don't really have any other argument. It also avoids the function call 🤷 

FWIW, you can't replace the value of `V2`, as it is a private type, and it doesn't accept `nil`.

Reply:
👍🏽 All good then